miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -149,6 +164,9 @@ end
 #! - value is the value to increment the nonce by. value can be at most 2^32 - 1 otherwise this
 #!   procedure panics.
 export.incr_account_nonce
+    # check that this procedure was executed against the native account
+    exec.check_native_account
```

Comment:
For this, and other similar procedures, we should add a doc comment with the new error condition.